### PR TITLE
Fix #3503

### LINKED ISSUE #3503: When you press fly while sitting in vehicle makes the vehicle's physics unavailable
- Sit in the vehicle
- Press `f` to fly
- Press tilde to find the vehicle's collider

### PR DESCRIPTION
**Pull request by aryasalehi**
## Describe your changes
This is the fix for : https://github.com/webaverse/app/issues/3503
related : https://github.com/webaverse/app/pull/3510

The reason that the previous PR wasn't merged is because we want to have the ability to fly or jump out of a mount for example, but we don't want that behavior for a hoverboard
So we're going to add a a property in the metaversefile which disables/enables jump or fly for a vehicle 

## What are the steps for a QA tester to test this pull request?
Same steps as the issue

## Issue ticket number and link
https://github.com/webaverse/app/issues/3503

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I am not adding any irrelevant code or assets
- [x] I am only including the changes needed to implement the change
- [x] I have playtested and intentionally tried to find error cases but couldn't

### CODE REVIEW

File: game.js [hunk 1]
```diff
@@ -67,6 +67,18 @@ const _getGrabbedObject = i => {
   return result;
 };
 
+const _unwearAppIfHasSitComponent = (localPlayer) => {
+  const wearActions = Array.from(localPlayer.getActionsState()).filter(action => action.type === 'wear');
```
Comment (avaer) [inline game.js:71]: Why not `player.hasAction('wear')`?

Comment (aryasalehi) [reply game.js:71]: Makes sense , I'll use that 👍

File: game.js [hunk 2]
```diff
@@ -1398,6 +1410,9 @@ class GameManager extends EventTarget {
         type: 'fly',
         time: 0,
       };
+      if (localPlayer.hasAction('sit')) {
+       _unwearAppIfHasSitComponent(localPlayer);
```
Comment (avaer) [inline game.js:1414]: Spacing.

File: game.js [hunk 3]
```diff
@@ -67,6 +67,18 @@ const _getGrabbedObject = i => {
   return result;
 };
 
+const _unwearAppIfHasSitComponent = (localPlayer) => {
```
Comment (avaer) [inline game.js:70]: Argument should be `player`.

File: game.js [hunk 4]
```diff
@@ -1398,6 +1410,9 @@ class GameManager extends EventTarget {
         type: 'fly',
         time: 0,
       };
+      if (localPlayer.hasAction('sit')) {
```
Comment (avaer) [inline game.js:1413]: This check seems pointless?

**Review by avaer — COMMENTED** [review]
Minor code cleanup, otherwise looks good.

File: game.js (hunk 1 above)
Comment (aryasalehi) [reply game.js:71]: It's actually player.getAction('wear')

File: game.js [hunk 5]
```diff
@@ -67,6 +67,16 @@ const _getGrabbedObject = i => {
   return result;
 };
 
+const _unwearAppIfHasSitComponent = (player) => {
+  const wearAction = player.getAction('wear');
```
Comment (avaer) [inline game.js:71]: Is this only handling the first wear action? Is that correct?

File: game.js [hunk 6]
```diff
@@ -67,6 +67,16 @@ const _getGrabbedObject = i => {
...
+  const instanceId = wearAction.instanceId;
+  const app = metaversefileApi.getAppByInstanceId(instanceId);
+  const sitComponent = app.getComponent('sit');
```
Comment (avaer) [inline game.js:74]: This could be `hasComponent` but it doesn't matter much.

File: character-controller.js [hunk 7]
```diff
@@ -749,6 +749,11 @@ class StatePlayer extends PlayerBase {
       this.characterPhysics.setPosition(position);
     }
   }
+  getActionsWithType(type) {
```
Comment (avaer) [inline character-controller.js:752]: I think `getActionsByType` might be more in line with current naming schemes?

**Review by avaer — APPROVED** [review]

**Pull request merged**